pkg/fs_algo/fs_algo/fs_perf_viz.py

```diff
@@ -0,0 +1,428 @@
+                    plt.xlabel('Latitude',fontsize = 26)
+                    plt.ylabel('Longitude',fontsize = 26)
+                    cbar_ax = plt.colorbar(cbar, ax=ax,fraction=0.02, pad=0.04)
+                    cbar_ax.set_label(label=metric,size=24)
+                    cbar_ax.ax.tick_params(labelsize=24)  # Set colorbar tick labels size
+                    plt.title("Predicted Performance: {}".format(ds), fontsize = 28)
+
+                    # Save the plot as a .png file
+                    output_path = f'{dir_out}/data_visualizations/{ds}_{algo}_{metric}_performance_map.png'
+                    plt.savefig(output_path, dpi=300, bbox_inches='tight')
+                    plt.clf()
+                    plt.close()
+
+                if 'obs_vs_sim_scatter' in true_keys:
+                    # Scatter plot of observed vs. predicted module performance
+                    # Remove 'USGS-' from ids so it can be merged with the actual performance data
+                    data['identifier'] = data['identifier'].str.replace(r'\D', '', regex=True)
+                    data['identifier'] = data['identifier'].str.strip() # remove leading and trailing spaces
+
+                    # Read in the observed performance data
+                    path_obs_perf = f'{dir_std_base}/{ds}/{ds}_{formulation_id}.{save_type_obs}'
+                    obs = xr.open_dataset(path_obs_perf, engine=engine)
+                    # NOTE: Below is one option, but it assumes there is only one possible .nc or .zarr file to read in (it only reads the first one it finds with that file extension)
+                    # obs = fsate._open_response_data_fs(dir_std_base=dir_std_base, ds=ds)
+                    obs = obs.to_dataframe()
+
+                    # Standardize column names
+                    obs.reset_index(inplace=True)
+                    obs = obs.rename(columns={"gage_id": "identifier"})
+
+                    # Subset columns
+                    data = data[['identifier', 'comid', 'X', 'Y', 'prediction', 'metric', 'dataset']]
+                    data = data[data['metric'] == metric]
+                    data.columns = data.columns.str.lower()
+                    obs = obs[['identifier', metric]]
+
+                    # Merge the observed and predicted data
+                    data = pd.merge(data, obs, how = 'inner', on = 'identifier')
+
+                    # Plot the observed vs. predicted module performance
+                    plt.scatter(data['prediction'], data[metric], c='teal')
+                    plt.axline((0, 0), (1, 1), color='black', linestyle='--')
+                    plt.xlabel('Predicted {}'.format(metric))
+                    plt.ylabel('Actual {}'.format(metric))
+                    plt.title('Observed vs. Predicted Performance: {}'.format(ds))
+
+                    # Save the plot as a .png file
+                    output_path = f'{dir_out}/data_visualizations/{ds}_{algo}_{metric}_obs_vs_sim_scatter.png'
```

Comment:
@bolotinl A new suggestion - create a subdirectory based on the dataset inside output/data_visualizations, mimicking the structures created by `fs_algo_train_eval.std_pred_path`, e.g. 
```
Path(dir_out)/Path('data_visualizations')/Path(dataset_id)/Path(insert specific plottype_algo_metric_dataset_id here)
```